packages/block-editor/src/components/rich-text/index.js

```diff
@@ -482,19 +487,30 @@ class RichTextWrapper extends Component {
 
 const RichTextContainer = compose( [
 	withInstanceId,
-	withBlockEditContext( ( { clientId } ) => ( { clientId } ) ),
+	withBlockEditContext( ( { clientId, onCaretVerticalPositionChange, isSelected }, ownProps ) => {
+		if ( Platform.OS === 'web' ) {
```

Comment:
An idea to keep the current `.native` system: create the higher order component in a separate fire where it returns the same component for web, and import it here. Native could have two `withBlockEditContext` higher order components.